cl/recap/tasks.py

```diff
@@ -205,26 +204,19 @@ def mark_pq_status(pq, msg, status, message_property_name="error_message"):
         logger.info(msg)
     setattr(pq, message_property_name, msg)
     pq.status = status
-    pq.save()
+    await pq.asave()
     return pq.status, getattr(pq, message_property_name)
 
 
-@app.task(
-    bind=True,
-    autoretry_for=(requests.ConnectionError, requests.ReadTimeout),
-    max_retries=5,
-    interval_start=5 * 60,
-    interval_step=10 * 60,
-)
-def process_recap_pdf(self, pk):
+async def process_recap_pdf(pk):
```

Comment:
OK, I'm going to try this one more time, just so you can tell me that, "Yes, Mike, this is fine."

I have two concerns here:

1. process_recap_pdf can do OCR and OCR can take a *really* long time. The longest is something like a day, but generally it's done within a minute or two. Sometimes it can take a lot longer though.  

    Given that that's the case, can we count on a client being *around* to receive a response? And if the client wanders away, could that screw up the processing? 

    In other words, if this method takes an hour and the user throws their laptop into a lake after a few seconds, is that OK?

2. We have two types of users for the RECAP upload API. The extension (via browsers), and API users. The API users are used to getting nearly instant responses, and this will make their responses *much* slower. Won't that screw up the speed that clients can do uploads via servers, if they don't currently have a multi-threaded way of uploading? That's a big change in how this API works.

Reply:
> 1. Given that that's the case, can we count on a client being _around_ to receive a response? And if the client wanders away, could that screw up the processing?

I think it should be ok.

> In other words, if this method takes an hour and the user throws their laptop into a lake after a few seconds, is that OK?

Probably no worse than the current situation where processing can fail somewhat randomly even when there is a success response to the initial upload, moving OCR out of the request path and into background job can always be done if it's an issue.

> The API users are used to getting nearly instant responses, and this will make their responses much slower.

Well right now the upload responses are mostly useless AFAIU, they don't accurately indicate if the request was successful. Having accurate response status codes is going to be something API users are likely to want.

> Won't that screw up the speed that clients can do uploads via servers, if they don't currently have a multi-threaded way of uploading?

It depends on the exact implementation of the upload client, keep in mind not having useful responses is problematic in general for API users.